testing/object_api/test_object_api_c.cpp

```diff
@@ -0,0 +1,230 @@
+    system_->finalize();
+  }
+
+  test_platform platform_;
+  test_system *system_;
+};
+
+/**
+ * @test       test_print_error
+ * @brief      Test: print_erro
+ * @details    When params to print_err is valid, it displays the std error.<br>
+ */
+TEST(object_api_c, test_print_err){
+  const std::string str = "Invalid string";
+  print_err(str.c_str(), FPGA_INVALID_PARAM);
+}
+
+/**
+ * @test       parse_args0
+ * @brief      Test: parse_args
+ * @details    When passed an invalid command option,<br>
+ *             parse_args prints a message and<br>
+ *             returns a value other than FPGA_OK.<br>
+ */
+TEST_P(object_api_c_p, parse_args0) {
+  char zero[20];
+  char one[20];
+  strcpy(zero, "object_api");
+  strcpy(one, "-Y");
+
+  char *argv[] = { zero, one };
+
+  EXPECT_NE(parse_args(2, argv), FPGA_OK);
+}
+
+/**
+ * @test       parse_args1
+ * @brief      Test: parse_args
+ * @details    When given a command option that requires a param,<br>
+ *             omitting the required param causes parse_args to<br>
+ *             return a value other than FPGA_OK.<br>
+ */
+TEST_P(object_api_c_p, parse_args1) {
+  char zero[20];
+  char one[20];
+  char two[20];
+  strcpy(zero, "object_api");
+  strcpy(one, "-B");
+
+  char *argv[] = { zero, one, two };
+
+  EXPECT_NE(parse_args(3, argv), FPGA_OK);
+}
+
+/**
+ * @test       parse_args2
+ * @brief      Test: parse_args
+ * @details    When given invalid command options,<br>
+ *             parse_args returns FPGA_EXCEPTION.<br>
+ */
+TEST_P(object_api_c_p, parse_args2) {
+  char zero[20];
+  char one[20];
+  char two[20];
+  char three[20];
+  strcpy(zero, "object_api");
+  strcpy(one, "-B");
+  strcpy(two, "3");
+  strcpy(three, "-s");
+
+  char *argv[] = { zero, one, two, three };
+
+  EXPECT_EQ(parse_args(4, argv), FPGA_EXCEPTION);
+}
+
+/**
+ * @test       parse_args3
+ * @brief      Test: parse_args
+ * @details    When given valid command options,<br>
+ *             parse_args populates the global config struct,<br>
+ *             and returns FPGA_OK.<br>
+ */
+TEST_P(object_api_c_p, parse_args3) {
+  char zero[20];
+  char one[20];
+  char two[20];
+  char three[20];
+  strcpy(zero, "object_api");
+  strcpy(one, "-B");
+  strcpy(two, "3");
+
+  char *argv[] = { zero, one, two };
+
+  EXPECT_EQ(parse_args(3, argv), FPGA_OK);
+}
+
+/**
+ * @test       main0
+ * @brief      Test: hello_fpga_main
+ * @details    When given an invalid command option,<br>
+ *             object_api displays an error message,<br>
+ *             and returns non-zero.<br>
+ */
+TEST_P(object_api_c_p, main0) {
+  char zero[20];
+  char one[20];
+  strcpy(zero, "object_api");
+  strcpy(one, "-Y");
+
+  char *argv[] = { zero, one };
+
+  EXPECT_NE(object_api_main(2, argv), 0);
+}
+
+INSTANTIATE_TEST_CASE_P(object_api_c, object_api_c_p,
+                        ::testing::ValuesIn(test_platform::platforms({"skx-p"})));
+
+
+class object_api_c_mock_p : public ::testing::TestWithParam<std::string> {
```

Comment:
Is this class exactly the same as the previous fixture? Can we just inherit from the `object_api_c_p` instead?